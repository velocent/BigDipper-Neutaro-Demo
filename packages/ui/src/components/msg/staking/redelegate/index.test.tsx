import renderer from 'react-test-renderer';
import Redelegate from '@/components/msg/staking/redelegate';
import { MsgRedelegate } from '@/models';
import MockTheme from '@/tests/mocks/MockTheme';

// ==================================
// mocks
// ==================================
jest.mock('@/components/name', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Name" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/Redelegate', () => {
  it('matches snapshot', () => {
    const message = MsgRedelegate.fromJson({
      category: 'staking',
      type: 'MsgEditValidator',
      delegator_address: 'delegatorAddress',
      validator_src_address: 'validatorSrcAddress',
      validator_dst_address: 'validatorDstAddress',
      amount: {
        denom: 'udaric',
        amount: '1000000000',
      },
    });
    const component = renderer.create(
      <MockTheme>
        <Redelegate message={message} />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
