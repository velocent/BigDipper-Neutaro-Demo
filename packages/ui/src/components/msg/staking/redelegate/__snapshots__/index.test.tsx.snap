// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/Redelegate matches snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<p
  className="MuiTypography-root MuiTypography-body1 emotion-0"
>
  <span
    data-testid="Trans"
    i18nKey="message_contents:txRedelegateContent"
  >
    <div
      address="delegatorAddress"
      id="Name"
      name="delegatorAddress"
    />
    <b />
    <div
      address="validatorSrcAddress"
      id="Name"
      name="validatorSrcAddress"
    />
    <div
      address="validatorDstAddress"
      id="Name"
      name="validatorDstAddress"
    />
    {"amount":"1,000 DARIC"}
  </span>
</p>
`;
